File: Codigos estudiantes por lenguaje/C++/Cristian Maldonado/Geometria/Heron_Formula.cpp
#include <iostream>
#include <cmath>

using namespace std;

typedef long double ld;

ld heron(ld x1, ld y1, ld x2, ld y2, ld x3, ld y3){
    ld a = sqrt((x1 - x2)*(x1 - x2)+(y1 - y2)*(y1 - y2));
    ld b = sqrt((x1 - x3)*(x1 - x3)+(y1 - y3)*(y1 - y3));
    ld c = sqrt((x3 - x2)*(x3 - x2)+(y3 - y2)*(y3 - y2));
    ld s = (a + b + c)/2.0;
    ld A = sqrt(s *(s - a)*(s - b)*(s - c));
    return A;
}

int main()
{
    ios_base::sync_with_stdio(false);cin.tie(NULL);cout.tie(NULL);
    cout.precision(22);
    ld x1, y1, x2, y2, x3, y3; cin>>x1>>y1>>x2>>y2>>x3>>y3;
    cout<<heron(x1, y1, x2, y2, x3, y3)<<endl;
    return 0;
}

File: Codigos estudiantes por lenguaje/Java/Cristian Maldonado/Geometria/Heron_Formula.java
public class Heron_Formula {

    public static void main(String[] args) {
        System.out.println(heron(1, 0, -1, 0, 0, 2));
    }

    static double heron(double x1, double y1, double x2,
            double y2, double x3, double y3) {
        double a = Math.sqrt((x1 - x2) * (x1 - x2) + (y1 - y2) * (y1 - y2));
        double b = Math.sqrt((x1 - x3) * (x1 - x3) + (y1 - y3) * (y1 - y3));
        double c = Math.sqrt((x3 - x2) * (x3 - x2) + (y3 - y2) * (y3 - y2));
        double s = (a + b + c) / 2.0;
        double A = Math.sqrt(s * (s - a) * (s - b) * (s - c));
        return A;
    }
}
